docs/how-to/deploy_sdcore_cups.md

```diff
@@ -8,70 +8,237 @@ This guide covers how to install a SD-Core 5G core network with Control Plane an
+cd terraform-juju-sdcore-k8s/modules/sdcore-control-plane-k8s
+cat << EOF > control-plane.tfvars
+model_name = "control-plane"
+create_model = false
+amf_config = {
+  external-amf-ip       = "10.201.0.201"
+  external-amf-hostname = "amf.core"
+}
+
 EOF
 ```
 
-Create a Juju model to represent the Control Plane.
+Initialize Juju Terraform provider:
 
 ```console
-juju add-model control-plane control-plane-cloud
+terraform init
 ```
 
-Deploy the control plane bundle.
+Deploy SD-Core Control Plane
+
 ```console
-juju deploy sdcore-control-plane-k8s --trust --channel=beta --overlay control-plane-overlay.yaml
+terraform apply -var-file="control-plane.tfvars" -auto-approve
 ```
 
-Expose the integration offer for the AMF N2 interface.
+### Integration with the AMF N2 interface
+
+The AMF charm allows establishing the N2-plane connectivity through the `fiveg_n2` charm interface. 
+
+``````{tab-set}
+
+`````{tab-item} Option 1: Integration within the same Juju model
+
+It is assumed that the `fiveg-n2` requirer application is already deployed in the Juju model.
+To create a `fiveg_n2` integration between the AMF and another application within the same Juju model, add the following section to the `main.tf` file in the `terraform-juju-sdcore-k8s/modules/sdcore-control-plane-k8s` directory:
 
 ```console
-juju offer control-plane.amf:fiveg-n2
+resource "juju_integration" "fiveg-n2" {
+  model = "control-plane"
+
+  application {
+    name     = module.amf.app_name
+    endpoint = module.amf.fiveg_n2_endpoint
+  }
+
+  application {
+    name     = <THE `fiveg-n2` REQUIRER APP>
+    endpoint = <THE `fiveg-n2` REQUIRER APP'S INTEGRATION ENDPOINT>
+  }
+}
 ```
 
-## Deploy SD-Core User Plane
+Apply the changes:
+
+```console
+terraform apply -var-file="control-plane.tfvars" -auto-approve
+```
+
+`````
+
+`````{tab-item} Option 2: Cross-model integration
 
-Create a Juju overlay file.
+In this option, it is assumed that the `sdcore-control-plane-k8s` has been deployed as part (a sub-module) of a bigger system.
+The `sdcore-control-plane-k8s` Terraform module exposes the AMF application name and the `fiveg-n2` endpoint through the `output.tf` file.
+To create a cross-model `fiveg_n2` integration in the root module of your deployment, add the following section to the `main.tf` file:
 
 ```console
-cat << EOF > upf-overlay.yaml
-applications:
-  upf:
-    options:
-      access-gateway-ip: 10.202.0.1
-      access-interface: access
-      access-ip: 10.202.0.10/24
-      core-gateway-ip: 10.203.0.1
-      core-interface: core
-      core-ip: 10.203.0.10/24
-      gnb-subnet: 10.204.0.0/24
-EOF
+resource "juju_offer" "amf-fiveg-n2" {
+  model            = "control-plane"
+  application_name = module.<CONTROL PLANE MODULE NAME>.amf_app_name
+  endpoint         = module.<CONTROL PLANE MODULE NAME>.fiveg_n2_endpoint
+}
+
+resource "juju_integration" "fiveg-n2" {
+  model = "control-plane"
+
+  application {
+    name     = <THE `fiveg-n2` REQUIRER APP>
```

Comment:
We do not mention about the deployment of fiveg-n2 requirer app (Gnbsim). It seemed to me that a piece is missing here it users only checks this document. We know that it's included in another how-to. What about giving a reference to  Gnbsim deployment how-to guide.

Reply:
The `fiveg-n2` requirer is not necessarily gNB simulator. We won't be using it at all in the real deployments. I don't think that the how-to guide should focus on this aspect.